eppo_client/client.py

```diff
@@ -47,6 +127,12 @@ def get_assignment(
         experiment_config = self.__config_requestor.get_configuration(flag_key)
         override = self._get_subject_variation_override(experiment_config, subject_key)
         if override:
+            if expected_variation_type is not None:
+                variation_is_expected_type = VariationType.is_expected_type(
```

Comment:
I like how you folded the "is expected type" logic into `VariationType` 👏 

Clean! 🧼